EpsilonLights/Tests/tests/LightsTest/LightsTest.c

```diff
@@ -0,0 +1,373 @@
+    driversInputs[1] = 0b00010000;
+    driversInputs[2] = 0b00000001;
+    auxBmsInputs[1] = 0b00000001;
+    lightsInfo.leftSignal = 0;
+    lightsInfo.rightSignal = 0;
+
+    osDelayUntil_ExpectAndReturn(&prevWakeTime, LIGHTS_UPDATE_FREQ, 0);
+    HAL_GPIO_WritePin_Expect(HHIGH_GPIO_Port, HHIGH_Pin, LIGHT_OFF);
+    HAL_GPIO_WritePin_Expect(HLOW_GPIO_Port, HLOW_Pin, LIGHT_ON);
+    HAL_GPIO_WritePin_Expect(BRAKE_GPIO_Port, BRAKE_Pin, LIGHT_ON);
+    HAL_GPIO_WritePin_Expect(ESTROBE_GPIO_Port, ESTROBE_Pin, LIGHT_ON);
+    updateLights(&lightsInfo, &prevWakeTime, &regenBrakeInfo);
+
+    TEST_ASSERT_EQUAL_MESSAGE(LIGHT_ON, sigLightsHandle.left,
+                              "sigLightsHandle.left expected to be LIGHT_ON");
+    TEST_ASSERT_EQUAL_MESSAGE(LIGHT_ON, sigLightsHandle.right,
+                              "sigLightsHandle.right expected to be LIGHT_ON");
+}
+
+/*test blinkSignalLights().Both signal lights are disabled. CHecking to see if
+prevSigState is 0 or not*/
+void test_blinkSignalLights_sigLightsLeftOFFSigLightsRightOFF()
+{
+    uint32_t prevWakeTime = 0;
+    uint8_t prevSigState = 10;
+    uint32_t blinkerTimer = 0;
+    sigLightsHandle.left = 0;
+    sigLightsHandle.right = 0;
+
+    osDelayUntil_ExpectAndReturn(&prevWakeTime, LIGHTS_UPDATE_FREQ, 0);
+    HAL_GPIO_WritePin_Expect(RSIGNAL_GPIO_Port, RSIGNAL_Pin, LIGHT_OFF);
+    HAL_GPIO_WritePin_Expect(LSIGNAL_GPIO_Port, LSIGNAL_Pin, LIGHT_OFF);
+    blinkSignalLights(&prevWakeTime, &blinkerTimer, &prevSigState);
+
+    TEST_ASSERT_EQUAL_MESSAGE(0, prevSigState, "prevSigState is supposed to be 0");
+}
+
+/*test blinkSignalLights. left signal and right signal are both on.
+Checking to see if blinkerTimer becomes 0 and if prevSigState becomes 1*/
+void test_blinkSignalLights_sigLightsLeftONSigLightsRightONPrevSigStateZero()
+{
+    uint32_t prevWakeTime = 0;
+    uint8_t prevSigState = 0;
+    uint32_t blinkerTimer = 0;
+    sigLightsHandle.left = 1;
+    sigLightsHandle.right = 1;
+    osDelayUntil_ExpectAndReturn(&prevWakeTime, LIGHTS_UPDATE_FREQ, 0);
+    HAL_GPIO_WritePin_Expect(RSIGNAL_GPIO_Port, RSIGNAL_Pin, sigLightsHandle.right);
+    HAL_GPIO_WritePin_Expect(LSIGNAL_GPIO_Port, LSIGNAL_Pin, sigLightsHandle.left);
+    blinkSignalLights(&prevWakeTime, &blinkerTimer, &prevSigState);
+
+    TEST_ASSERT_EQUAL_MESSAGE(0, blinkerTimer, "blinkerTimer is supposed to be 0");
+    TEST_ASSERT_EQUAL_MESSAGE(1, prevSigState, "prevSigState is supposed to be 1");
+}
+
+/*test blinkSignalLights. left and right signals are both on. 
+prevSigState is 1.*/
+void test_blinkSignalLights_prevSigState1SigLightsLeftONSigLightsRightON()
+{
+    uint32_t prevWakeTime = 0;
+    uint8_t prevSigState = 1;
+    sigLightsHandle.left = 2;
+    sigLightsHandle.right = 2;
+
+    keepBlinkersOFF_resetBlinkerTimerToZero(prevWakeTime, prevSigState);
+    turnBlinkersON_updateTimer(prevWakeTime, prevSigState);
+    keepBlinkersOFF_updateTimer(prevWakeTime, prevSigState);
```

Comment:
If it's possible, we should try to have these in separate tests instead of having them all nested in here.

Reply:
I was wondering If we could talk about this tom. I just want to make sure that this is what you really want us to do

Reply:
nvm, just made them into separate test files lol